### PR TITLE
kata-sys-util: protection: Update TDX checks

### PR DESCRIPTION
**Pull request by jodh-intel**
Update the protection checking code to detect newer versions of Intel TDX (whose userland interface has now stabilised).

> **Note:** that we don't need to retain the existing behaviour since:
>
> - We haven't yet landed the TDX feature (#6448).
> - Systems wishing to use TDX will need to use the latest available
>   system components (such as firmware and host kernel).

Also added an explicit TDX unit test.

Fixes: #7384.

### CODE REVIEW

File: src/libs/kata-sys-util/src/protection.rs [hunk 1]
```diff
@@ -72,42 +70,20 @@ pub fn available_guest_protection() -> Result<GuestProtection, ProtectionError>
...
 }
 
-#[cfg(target_arch = "x86_64")]
-fn retrieve_cpu_flags() -> Result<String> {
-    let cpu_info =
-        crate::cpu::get_single_cpu_info(crate::cpu::PROC_CPUINFO, crate::cpu::CPUINFO_DELIMITER)?;
-
-    let cpu_flags =
-        crate::cpu::get_cpu_flags(&cpu_info, crate::cpu::CPUINFO_FLAGS_TAG).map_err(|e| {
-            anyhow!(
```
Comment (fidencio) [inline src/libs/kata-sys-util/src/protection.rs:88]: As you removed this one, the static checks are complaining about anyhow imported but not used.

**Review by fidencio — APPROVED** [review]
lgtm, assuming green tests.
Thanks, @jodh-intel!

**Comment by jodh-intel** [comment]
alpinelinux.org seems to have gone down that caused the tests to fail.

/retest

**Review by jiangliu — APPROVED** [review]

**Pull request merged**